src/software/guides/addin-icon-validator.html

```diff
@@ -0,0 +1,325 @@
+                    reader.readAsText(file);
+                    
+                    var selected = false;
+                    var errors = "";
+
+                    //CHECKING FILE SIZE
+                    if(file.size > 1024000){
+                        errors += "Exceeds the limit of 1024 kb <br>";
+                    }
+
+                    //DECLARING SVG FOR FILE TYPE CHECK
+                    var svgName = "image/svg+xml";
+
+                    //CHECKING FILE TYPE
+                    if(file.type != svgName){
+                        errors += "Not an SVG file<br>";
+                    }
+
+                    //CHOOSING DISPLAY TYPE
+                    if(document.getElementById('selected').checked){
+                        selected = true;
+                    }else if(document.getElementById('neutral').checked){
+                        selected = false;
+                    }
+
+                    reader.onload = function(e) {
+
+                        var rawLog = reader.result;
+                        
+                        //CHECKING SVG PROPERTIES FOR ERRORS
+                        if(rawLog.includes("viewBox=\"0 0 24 24\"") || (rawLog.includes("height=24") && rawLog.includes("width=24"))){
+                            errors += "";
+                        }else{
+                            errors += "Improper aspect ratio or dimensions. Must be 24 px square.<br>";
+                        }
+
+                        var preArray1 = rawLog.replace("<style>", ";");
+                        var preArray2 = preArray1.replace("</style>", ";");
+                        var preArray3 = preArray2.split("{");
+
+                        var array = preArray3.join(";").split(";");
+                        var text = "<p>";
+                        var i;
+
+                        for(i = 0; i < array.length; i++){
+                            text += array[i] + "<br>" ;
+
+                            var attr = array[i].split(":");
+
+                            if (attr.length > 1){
+                                var k = attr[0];
+                                var v = attr[1];
+                            }
+                            if (k == "stroke-width" && v != "2px"){
+                                errors += "The stroke width is " + v + "px, while it should be 2px. <br>";
+                            }
+                            if (k == "fill"){
+                                break;
+                            }
+                        }
+
+                        if(errors == ""){
+                            errors += "There are no issues with the uploaded file. <br>";
+                        }
+
+                        text += "</p>";
+
+                        //PRINTING THE ERRORS ON THE SCREEN
+                        document.body.insertAdjacentHTML("beforeend", "<div class=\"container\" id=\"errors\"><h2>Errors:</h2>" + errors + "<br></div>"); 
```

Comment:
Could we build a logic to not have the h2 element "Errors" when there are no issues with the uploaded file?

Reply:
added an if statement around the document.body.insertAdjacentHTML line to only display when errors != ""